src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -1209,7 +1209,7 @@ private void updateOverallHappiness()
             final BuildingHome home = citizen.getHomeBuilding();
             if(home != null)
             {
-                housing = home.getBuildingLevel();
+                housing += home.getBuildingLevel();
```

Comment:
So this would turn that into 
```
If you have 3 Citizens but only 1 Citizens' hut level 2 (with Citizen A and B in it)
You would have:
Citizen A +2
Citizen B +0 (Even though he is living in a house of level 2)
Citizen C +0
```